### PR TITLE
Add Sink Only Config to sink sample

### PR DESCRIPTION
**Pull request by sambhurst**

### CODE REVIEW

File: samples/subsys/usb_c/sink/src/main.c [hunk 1]
```diff
@@ -16,7 +16,8 @@ LOG_MODULE_REGISTER(main, LOG_LEVEL_DBG);
 #define PORT1_NODE DT_NODELABEL(port1)
 #define PORT1_POWER_ROLE		DT_ENUM_IDX(DT_NODELABEL(port1), power_role)
 
-#if (PORT1_POWER_ROLE != TC_ROLE_SINK)
+/* A Sink power role evauates to 0. See usbc_tc.h: TC_ROLE_SINK */
+#if (PORT1_POWER_ROLE)
```
Comment (keith-zephyr) [inline samples/subsys/usb_c/sink/src/main.c:20]: Why is this change needed? This check will fail if the `enum tc_power_role` options change.

File: samples/subsys/usb_c/sink/src/main.c [hunk 2]
```diff
@@ -16,7 +16,8 @@ LOG_MODULE_REGISTER(main, LOG_LEVEL_DBG);
 #define PORT1_NODE DT_NODELABEL(port1)
 #define PORT1_POWER_ROLE		DT_ENUM_IDX(DT_NODELABEL(port1), power_role)
 
-#if (PORT1_POWER_ROLE != TC_ROLE_SINK)
+/* A Sink power role evauates to 0. See usbc_tc.h: TC_ROLE_SINK */
```
Comment (alevkoy) [inline samples/subsys/usb_c/sink/src/main.c:19]: evaluates

File: samples/subsys/usb_c/sink/src/main.c (hunk 1 above)
Comment (alevkoy) [reply samples/subsys/usb_c/sink/src/main.c:20]: I am also curious about this.

Comment (sambhurst) [reply samples/subsys/usb_c/sink/src/main.c:20]: I broke something and thought this was the problem but it's okay. I'm going to abandon this PR.

File: samples/subsys/usb_c/sink/src/main.c (hunk 2 above)
Comment (sambhurst) [reply samples/subsys/usb_c/sink/src/main.c:19]: Yes it does.

**Review by alevkoy — DISMISSED** [review]
Perhaps the commit message should explain why you're doing this. Otherwise, seems fine.

**Comment by sambhurst** [comment]
> Perhaps the commit message should explain why you're doing this. Otherwise, seems fine.

@alevkoy Done

**Comment by alevkoy** [comment]
> > > > Perhaps the commit message should explain why you're doing this. Otherwise, seems fine.
> > > 
> > > 
> > > @alevkoy Done
> > 
> > 
> > Was it breaking before? Did you change how the configs work? Do you want to be more explicit now that there are more options?
> 
> @alevkoy No it wasn't broken. I added it because there are more options

Can you say that in the commit message? It still just says you're adding the config to this sample.

**Comment by sambhurst** [comment]
> 

@alevkoy I've updated the commit message.

**Review by alevkoy — APPROVED** [review]

**Review by keith-zephyr — APPROVED** [review]
